Update ILIAS Installation Documentation

fwolf-ilias opened the pull request:
Dear all,

We would like to invite you to take a look at the revised version of the installation guide docu/configuration/install.md.  The new version aims to streamline the steps needed to have a minimally functional ILIAS installation and additionally provide information on creating a more secure environment and more functions.  This should make it easier to get started without missing any information.  

If any information is missing or incorrectly displayed, please contact me in this PR or @fwolf-ilias . 
I am also happy to add links to other components that require further configuration, as it is the case with the chat, RPC server or plugins that are still missing.

Thank you very much!

Fabian Wolf 

**Summary of changes:**.
- Revised table of contents for better readability.
- Outdated information removed and installation steps streamlined.
- Updated and rearranged the recommended software setup and reference system details.
- Added a recommendation for installing ILIAS on Ubuntu 24.04 for a faster and easier initial installation
- Added a section on backing up ILIAS before performing minor and major upgrades.
- Organised optional configurations (e.g. WebDAV, security) in the appendix.
- Added more information about community involvement and release information.

fwolf-ilias committed:
Restructure ilias install.md to show the releavant installation steps more compressed together. Remove osolete data and rework some of the setps and informations

fwolf-ilias committed:
Rewrite Upgrade ILIAS

fwolf-ilias committed:
switch cli path

fwolf-ilias committed:
Rearrange software requirenments and reference system

fwolf-ilias committed:
Change the Introduction sentence of Supported software and reference system

acgruber commented on docs/configuration/install.md:
```diff
@@ -1,46 +1,39 @@
 
+This is the installation guide for ILIAS 10, providing step-by-step instructions to set up all necessary components, 
+including the web server, database, and ILIAS code. Follow these instructions carefully to ensure a successful 
+installation of the e-Learning software. Each section will guide you through the required configurations and setups 
```
```suggestion
installation of the e-learning software. Each section will guide you through the required configurations and setups 
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -61,105 +54,113 @@ ILIAS itself will use around **25GB**. From there you can calculate your storage
+## Supported Software Setup and Reference System
+
+The following software versions are required/supported for ILIAS 10. The table below lists these versions alongside the 
+current configuration of the ILIAS test server, which serves as a reference system (https://test10.ilias.de).
```
```suggestion
current configuration of the [ILIAS test server](https://test10.ilias.de), which serves as a reference system.
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -61,105 +54,113 @@ ILIAS itself will use around **25GB**. From there you can calculate your storage
-## Database Recommendations
+> Please note that other platforms and configurations should be possible, but it
+> might be harder to find someone who can help when things go south. You should not
+> use a different configuration unless you are an experienced system administrator.
```
```suggestion
> Please note that other platforms and configurations may be possible, but it
> might be difficult to find someone who can help in case of issues. We strongly recommend **not**
> using a different configuration unless you are an experienced system administrator.
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -61,105 +54,113 @@ ILIAS itself will use around **25GB**. From there you can calculate your storage
-We RECOMMEND to use MariaDB with the following settings:
+Depending on your Linux Distribution you have several ways to install the required
+dependencies. We recommend to always use your distributions package manager to
+keep your packages up to date in an easy manner avoiding security issues.
```
```suggestion
Depending on your Linux Distribution, you have several ways to install the required
dependencies. We recommend to always use your distributions package manager to
easily keep your packages up to date avoiding security issues.
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -61,105 +54,113 @@ ILIAS itself will use around **25GB**. From there you can calculate your storage
-  * `innodb_buffer_pool_size` > 2G (depending on DB size)
+Here we choose Ubuntu 24.04 because it already meets the recommended PHP version 8.3.
+For other Ubuntu or Debian-based Linux systems, we recommend using [https://deb.sury.org/](https://deb.sury.org/) to install the correct 
+PHP version later on.
```
```suggestion
In this guide we choose Ubuntu 24.04 because it already meets the recommended PHP version 8.3.
For other Ubuntu or Debian-based Linux systems, we recommend using [DEB.SURY.ORG](https://deb.sury.org/) to install the correct 
PHP version later on.
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -61,105 +54,113 @@ ILIAS itself will use around **25GB**. From there you can calculate your storage
-is set to `COMPACT`.
+`openjdk-17-jdk` and `maven` are optional and are used for the ILIAS RPC server for search indexing and certificate generation. 
+`git` is required if the source code is obtained directly from GitHub.
+Also `nodejs` and `npm` are required if you get the source directly to download the javascript dependencies in the installation process.
```
```suggestion
`nodejs` and `npm` are required as well if you get the source directly to download the javascript dependencies in the installation process.
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -61,105 +54,113 @@ ILIAS itself will use around **25GB**. From there you can calculate your storage
+`openjdk-17-jdk` and `maven` are optional and are used for the ILIAS RPC server for search indexing and certificate generation. 
+`git` is required if the source code is obtained directly from GitHub.
+Also `nodejs` and `npm` are required if you get the source directly to download the javascript dependencies in the installation process.
+Alternatively, they can be obtained directly from the distribution package at [https://deb.nodesource.com/](https://getcomposer.org/download/) to select appropriate nodejs versions according to the [Recommended Setup for Running ILIAS](#recommended-setup-for-running-ilias).
```
Which website link is correct here? There's 2 different ones in [] and (). Also, I would only put the website's title in the square brackets, not the full URL (only if necessary for clarification, I'd use a shortened version of the URL). So if Nodesource should be correct here, it would look like this: \[Nodesource\](https://deb.nodesource.com/).

acgruber commented on docs/configuration/install.md:
```diff
@@ -61,105 +54,113 @@ ILIAS itself will use around **25GB**. From there you can calculate your storage
 
-<a name="install-dependencies"></a>
-## Install Dependencies
+The PHP packet manager Composer is necessary to download all external PHP dependencies and for php autoloading. 
```
```suggestion
The PHP packet manager Composer is necessary to download all external PHP dependencies and for PHP autoloading. 
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -61,105 +54,113 @@ ILIAS itself will use around **25GB**. From there you can calculate your storage
-<a name="install-dependencies"></a>
-## Install Dependencies
+The PHP packet manager Composer is necessary to download all external PHP dependencies and for php autoloading. 
+Alternatively it can be obtained directly from [https://getcomposer.org/download/](https://getcomposer.org/download/). 
```
```suggestion
Alternatively, it can be obtained directly from [getcomposer.org](https://getcomposer.org/download/). 
```

acgruber commented on docs/configuration/install.md:
````diff
@@ -61,105 +54,113 @@ ILIAS itself will use around **25GB**. From there you can calculate your storage
-On RHEL/CentOS execute:
-```
-yum install httpd
+Also to store the ilias configuartion which is later used to configurate ilias we create the folder /var/www/config . to prevent problems later with npm we create /var/www/.npm with webowner rights.
````
```suggestion
Also, to store the ILIAS configuration, which is later used to configurate ILIAS, we create the folder /var/www/config. To prevent future issues with npm we create /var/www/.npm with webowner rights.
```

acgruber commented on docs/configuration/install.md:
````diff
@@ -174,54 +175,29 @@ on Debian). A minimal configuration may look as follows:
-`/etc/php5/apache2/php.ini`). Delete the file `phpinfo.php` afterwards.
+Then point your browser to ```http://example.com/phpinfo.php```. 
+If you see the content of the file as shown above your configuration is **not** working. 
+If you can see details of your PHP Configuration everything works fine. 
````
```suggestion
If you see the content of the file as shown above, your configuration is **not** working. 
If you can see details of your PHP configuration, everything works fine. 
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -174,54 +175,29 @@ on Debian). A minimal configuration may look as follows:
+Delete the file `phpinfo.php` afterwards.
 
-We recommend the following settings for your php.ini:
+We recommend the atleast following settings for your php.ini:
```
```suggestion
We recommend at least the following settings for your php.ini:
```

acgruber commented on docs/configuration/install.md:
````diff
@@ -252,113 +228,60 @@ max_input_vars = 10000
-columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
-```
+Please ensure that PHP is compiled with `libargon2`. 
+This is mostly the case for common distributions, but if you compile PHP yourself it must be build with `--with-password-argon2[=DIR]` (see: https://www.php.net/manual/en/password.installation.php).
````
```suggestion
This is mostly the case for common distributions, but if you compile PHP yourself, it must be build with `--with-password-argon2[=DIR]` (see: https://www.php.net/manual/en/password.installation.php).
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -252,113 +228,60 @@ max_input_vars = 10000
+> ILIAS supports utf8-collations with 3 bytes per character, such as `utf8_general_ci`, only.
 
-If strict mode is disabled, there will be no output.
+We RECOMMEND to use MariaDB with the following settings:
```
```suggestion
We **strongly recommend** to use MariaDB with the following settings:
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -369,51 +292,41 @@ the update to future releases and versions.
-The commands above will directly serve ILIAS from the docroot.
+We recommend that the rights of the ILIAS code in the production system are restricted so that the web server only has 
+read access to the code. For this and other important security considerations, please refer to the security
+instructions in the [security guide](./secure.md).
```
```suggestion
We recommend restricting the rights of the ILIAS code in the production system so that the web server only has 
read access to the code. For this and other important security considerations, please refer to the security
instructions in the [Security Guide](./secure.md).
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -422,265 +335,250 @@ the [ILIAS Setup on the command-line](../../setup/README.md).
 or into the help of the program itself `php cli/setup.php help`. It is the tool
 to manage and monitor your ILIAS installation.
 
+If you are installing from git, it is possible that ILIAS will already require some few migration to the initial
```
```suggestion
If you are installing from Git, it is possible that ILIAS will already require a few migrations to the initial
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -422,265 +335,250 @@ the [ILIAS Setup on the command-line](../../setup/README.md).
+1. **ILIAS Java RPC server**
+It is used for certain optional functions such as Lucene Search
+or generating PDF Certificates. See [Lucene RPC-Server](../../components/ILIAS/WebServices/RPC/lib/README.md) for details
+on how to install the rpc server.
```
```suggestion
on how to install the RPC server.
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -422,265 +335,250 @@ the [ILIAS Setup on the command-line](../../setup/README.md).
-To develop plugins, you can start in our [development guide](http://www.ilias.de/docu/goto.php?target=st_27029).
-A variety of free plugins is provided from our community via the [ILIAS Plugin Repository](http://www.ilias.de/docu/goto.php?target=cat_1442&client_id=docu).
+A variety of free plugins is provided from our community via the [ILIAS Plugin Repository](https://docu.ilias.de/go/cat/1442).
+To develop plugins, you can start in our [development guide](https://docu.ilias.de/go/pg/27030_42).
```
```suggestion
To develop plugins, you can get started in our [Development Guide](https://docu.ilias.de/go/pg/27030_42).
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -422,265 +335,250 @@ the [ILIAS Setup on the command-line](../../setup/README.md).
 # Upgrading ILIAS
 
 The easiest way to update ILIAS is using git, please note that this is only possible
 if you installed ILIAS via git as advised in this document. If git wasn't used you
```
```suggestion
if you installed ILIAS via Git as advised in this document. If Git wasn't used you
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -422,265 +335,250 @@ the [ILIAS Setup on the command-line](../../setup/README.md).
 <a name="upgrading-ilias"></a>
 # Upgrading ILIAS
 
 The easiest way to update ILIAS is using git, please note that this is only possible
```
```suggestion
The easiest way to update ILIAS is using Git. Please note that this is only possible
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -422,265 +335,250 @@ the [ILIAS Setup on the command-line](../../setup/README.md).
+</details>
 
-In case of merge conflicts refer to [the ILIAS Development Guide](http://www.ilias.de/docu/goto.php?target=pg_15604). 
+In case of merge conflicts refer to [the ILIAS Development Guide](http://www.ilias.de/docu/goto.php?target=pg_15604).
```
```suggestion
In case of merge conflicts, refer to [the ILIAS Development Guide](http://www.ilias.de/docu/goto.php?target=pg_15604).
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -422,265 +335,250 @@ the [ILIAS Setup on the command-line](../../setup/README.md).
 ## Major Upgrade
 
-To apply a major update (e.g. v6.13 to 7.0) please check that your OS has the
+To apply a major update (e.g. v9.13 to v10.1) please check that your OS has the
```
```suggestion
To apply a major upgrade (e.g. v9.13 to v10.1) please check that your OS has the
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -422,265 +335,250 @@ the [ILIAS Setup on the command-line](../../setup/README.md).
-To apply a major update (e.g. v6.13 to 7.0) please check that your OS has the
+To apply a major update (e.g. v9.13 to v10.1) please check that your OS has the
 [proper dependency versions](#upgrading-dependencies) installed. If everything
 is fine change your default skin to Delos and apply this change at least to
```
```suggestion
is fine, change your default skin to Delos and apply this change at least to
```

acgruber commented on docs/configuration/install.md:
````diff
@@ -422,265 +335,250 @@ the [ILIAS Setup on the command-line](../../setup/README.md).
+```
 
-Complete the update of the base system by [updating the database](#database-update).
+Then update the code of your plugins according to their documentation so that it is compatible with the new ILIAS version.
````
```suggestion
Then update the code of your plugins according to their documentation to ensure they are compatible with the new ILIAS version.
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -422,265 +335,250 @@ the [ILIAS Setup on the command-line](../../setup/README.md).
+
+Complete the update of the base system by [updating the database](#database-the-update).
 
 As a last step you should log in with a User using your custom skin. If everything
 works fine change back from Delos to your custom system style. If not, you probably
 will need to update your style to match the new release.
```
```suggestion
As a last step, you should log in with a User using your custom skin. If everything
works fine, change back from Delos to your custom system style. If not, you probably
will need to update your style to match the new release.
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -714,39 +610,54 @@ questions. You might want to have a look into the [documentation of the command
 To keep your ILIAS Installation secure and healthy it is important that you keep
 it up to date. To get informations about updates and security fixes you should
-consider to subscribe to the [ILIAS Admin Mailing-List](http://lists.ilias.de/cgi-bin/mailman/listinfo/ilias-admins)
+consider to subscribe to the [ILIAS Admin Mailing-List](http://lists.ilias.de/cgi-bin/mailman/listinfo/ilias-admins). Information on the 
```
```suggestion
consider subscribing to the [ILIAS Admin Mailing-List](http://lists.ilias.de/cgi-bin/mailman/listinfo/ilias-admins). Information on the 
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -714,39 +610,54 @@ questions. You might want to have a look into the [documentation of the command
-consider to subscribe to the [ILIAS Admin Mailing-List](http://lists.ilias.de/cgi-bin/mailman/listinfo/ilias-admins)
+consider to subscribe to the [ILIAS Admin Mailing-List](http://lists.ilias.de/cgi-bin/mailman/listinfo/ilias-admins). Information on the 
+new versions, such as Important Changes, Known Issues, Changed Behaviour and Fixed 
+Issues, can be found in the release notes [Download & Releases](https://docu.ilias.de/go/lm/35).
```
```suggestion
Issues, can be found in the release notes in [Download & Releases](https://docu.ilias.de/go/lm/35).
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -755,43 +666,60 @@ each ILIAS release.
-We are also looking for [contributions of code](../development/contributing.md),
-[reports of issues](http://mantis.ilias.de) or [requests in our Feature Wiki](https://docu.ilias.de/goto.php?target=wiki_5307&client_id=docu#ilPageTocA119).
+We recommend to perform a threat analysis for your ILIAS installation, as every
+prudent admin should do for his resources. In our [security guide](./secure.md)
```
```suggestion
prudent admin should do for his resources. In our [Security Guide](./secure.md)
```

acgruber commented on docs/configuration/install.md:
````diff
@@ -755,43 +666,60 @@ each ILIAS release.
+```
+
+As a workaround `STRICT_TRANS_TABLES`, `STRICT_ALL_TABLES` and `ONLY_FULL_GROUP_BY`
+must be disabled. To do so create the file `/etc/mysql/conf.d/disable_strict_mode.cnf`
````
```suggestion
must be disabled. To do so, create the file `/etc/mysql/conf.d/disable_strict_mode.cnf`
```

acgruber reviewed (changes requested):
Hello @fwolf-ilias,

Thanks a lot for a contribution that will be incredibly helpful for our community!

Just completed my review. I’ve mainly fixed some minor typos and made suggestions to polish the language style a bit.

Let me know if you'd like to discuss anything! :)
Cheers!

fwolf-ilias committed:
Apply suggestions from @acgruber's code review

I appreciate you for taking the time to review my installation documentation! Your support means a lot—thank you! 😊

Co-authored-by: acgruber <116066877+acgruber@users.noreply.github.com>

fwolf-ilias committed:
Remove copy paste issue with the nodesource link. Thank you @acgruber

fwolf-ilias reviewed (commented):
I appreciate you for taking the time to review my changes to the guide! Your support means a lot—thank you! 😊

acgruber reviewed (approved):
Thanks a lot for implementing the changes, @fwolf-ilias! :)

mbecker-databay commented:
Lovely! But: This seems to miss the cronjob-setup.

fwolf-ilias commented:
@mbecker-databay You are absolutely right! I will definitely add that.

fwolf-ilias committed:
Add Cron Jobs to the install document.

fwolf-ilias committed:
Some details of cron execution and new headings in the appendix

acgruber commented on docs/configuration/install.md:
```diff
@@ -679,8 +683,36 @@ We strongly recommend using MariaDB instead of MySQL due to performance, licensi
+<a name="configurate-cron"></a>
+## Configure Cron Jobs
+
+This step configures the execution of the cron job, which can be set to perform tasks such as sending
```
```suggestion
This step configures the execution of the cron job, which can be set to perform tasks, such as sending
```

acgruber commented on docs/configuration/install.md:
```diff
@@ -679,8 +683,36 @@ We strongly recommend using MariaDB instead of MySQL due to performance, licensi
+The `<user>` MUST be a valid, arbitrary user account within the ILIAS installation.
+The `<client_id>` MUST correspond to the client ID of the ILIAS installation.
+
+To configure an automated cron job in your system, you need to create a user in ILIAS for example named `cron`.
```
```suggestion
To configure an automated cron job in your system, you need to create a user in ILIAS, for example named `cron`.
```

acgruber reviewed (approved):
checked new text

fwolf-ilias committed:
Apply suggestions from code review

Co-authored-by: acgruber <116066877+acgruber@users.noreply.github.com>

fwolf-ilias committed:
Wording.

PurHur commented on docs/configuration/install.md:
````diff
@@ -61,105 +55,113 @@ ILIAS itself will use around **25GB**. From there you can calculate your storage
-## Other Platforms or Configurations
+**Required PHP Extensions:**
+```
+gd, dom, xsl, mysql, curl, json, simplexml, libxml, xml, zip, imagick, intl, mbstring
````
pdo is missing.  (probably instead of mysql?)

fwolf-ilias committed:
Change mysql to pdo_mysql to reflect extensions instead of apt packages.

fwolf-ilias committed:
Add pdo because it is a dependncie of pdo_mysql and has to be installed in some systems like docker php and RHEL

ckozianka commented:
Great work @fwolf-ilias and @acgruber ! This will save a lot of people a lot of time 👍 

Maybe in the future you could consider two additional points:
- Virtual host configuration for Nginx (it has become more popular than Apache in the recent years)
- Providing advice on running composer as www-data when a lot of files are owned by root / not writable for www-data for security reasons (although the security guide might be the appropriate place for this)

Kind regards
Carsten

fwolf-ilias commented:
Hi @ckozianka,

thank you for your feedback!

**Configuration for Nginx:**
I would be happy to offer a configuration of nginx too. The reason why it is missing is, I rely soley on apache2. Since nobody wanted to maintain this document in the past, I agreed with the TB to do it on the condition that I only have to describe and maintain techniques for which my know-how is sufficient. 

So I'll be happy to receive PRs for Nginx support.

**Composer and File Permissions:**
I thought about that but I think this is out of scope for this instructions. There is a part where I recommend to restrict access rights and refer to the Security Guide:
> We recommend restricting the rights of the ILIAS code in the production system so that the web server only has read access to the code. For this and other important security considerations, please refer to the security instructions in the [Security Guide](https://github.com/ILIAS-eLearning/ILIAS/blob/12f985932fa8046a19f07f2b89d196e8c53d8bb9/docs/configuration/secure.md).

How to implement this in detail is up to the server administrator for the moment because there are plenty ways of doing that. We do not offer specific build processes in general because they vary widely over different systems and setups. But I could highlight the Recommendation more. 

Bests,

Fabian

fwolf-ilias committed:
Highlight important notes and fix some links

dsstrassner commented:
hi @fwolf-ilias could we remove the JF Label?
